### PR TITLE
Add reduce method

### PR DESCRIPTION
**Pull request by shuheiktgw**
I know it's not necessary to implement `reduce`, but as an functional programmer, I could not help implementing it haha.

### CODE REVIEW

File: vm/array_test.go [hunk 1]
```diff
@@ -752,6 +752,88 @@ func TestArrayPushMethod(t *testing.T) {
...
+			prev + n
+		end
+		`, "ArgumentError: Expect 0 or 1 argument. got=2"},
+	}
+
+	for i, tt := range testsFail {
+		v := initTestVM()
+		evaluated := v.testEval(t, tt.input)
+		checkError(t, i, evaluated, ArgumentError, tt.expected)
+		v.checkCFP(t, i, 2)
```
Comment (shuheiktgw) [inline vm/array_test.go]: I got cfp = 2 but I'm not sure if this is right or not... Could you tell me what you think bout this?

**Comment by shuheiktgw** [comment]
Hmm, I thought I covered all the code I wrote but travis tells me not...
Anyway could you review the code whenever you feel like it?

File: vm/array_test.go (hunk 1 above)
Comment (Maxwell-Alexius) [inline vm/array_test.go]: Try change to this :)
```go
 	evaluated := v.testEval(t, tt.input, getFilename())
 	checkError(t, i, evaluated, ArgumentError, tt.expected, getFilename(), tt.errorLine)
```

Comment (Maxwell-Alexius) [inline vm/array_test.go]: You may need help from @st0012 to explain the detail about CFP

Comment (shuheiktgw) [inline vm/array_test.go]: Oops, thanks Maxwell! I haven't noticed the change! I just fixed it!!

**Pull request merged**